### ISSUE TITLE
Trying to update docker desktop on windows causes and error.

### ISSUE THREAD

**Issue by HighDeFing** [docker/for-win#10451]
- [ ] I have tried with the latest version of Docker Desktop
  - [ ] I have tried disabling enabled experimental features
  - [x ] I have uploaded Diagnostics
  - Diagnostics ID:E6A6891D-FF20-4598-B529-874638452CC3/20210223033432

### Actual behavior
I was trying to update docker and then this error appeared also I got this error when trying to use ubuntu on windows https://askubuntu.com/questions/1142359/error-0x80080005-server-execution-failed-while-installing-ubuntu-on-windows

### Expected behavior
Update docker and use docker compose in windows 10 home.

### Information

  - Windows Version: Windows 10 Home 2004
  - Docker Desktop Version: 2.3.0.4
  - WSL2 or Hyper-V backend?  WSL2
  - Are you running inside a virtualized Windows e.g. on a cloud server or a VM: No, windows 10 home

### Steps to reproduce the behavior

  1. Just trying to update the docker image in desktop docker in windows 10 home 
  2. Also trying to install or use WSL2 or ubuntu on windows, I have "Windows Subsystem for Linux" active ass well and followed this things here with no avail https://askubuntu.com/questions/1142359/error-0x80080005-server-execution-failed-while-installing-ubuntu-on-windows

**Comment by stephen-turner** [comment]
If you haven't installed WSL 2 successfully, Docker will certainly not be able to run. I suggest you follow Microsoft's instructions at https://aka.ms/wsl2-install.

**Comment by HighDeFing** [comment]
I have it installed it just says I don't lol, thanks anyway. I will follow them microsoft instructions see if something works.

**Comment by stephen-turner** [comment]
If it's successfully installed, the command `wsl -l -v` should show some version 2 distros.

**Comment by HighDeFing** [comment]
Yes I get 
![image](https://user-images.githubusercontent.com/39177022/108880224-74544e80-75d8-11eb-840d-41d61f95d68b.png)

**Comment by stephen-turner** [comment]
That all looks good so I would expect it to work. What exactly are the steps to reproduce and what is the symptom?

**Comment by HighDeFing** [comment]
I just open docker desktop and go for an update, after a while 
![image](https://user-images.githubusercontent.com/39177022/108908889-3bc56c80-75fa-11eb-8716-7cce5f21161d.png) It just stops and shows me this too: 
![image](https://user-images.githubusercontent.com/39177022/108909283-be4e2c00-75fa-11eb-82c9-7d5b4d54e92a.png)
When I click `here` It takes me here: 
![image](https://user-images.githubusercontent.com/39177022/108909369-d756dd00-75fa-11eb-97c0-156f03930fdb.png)

Maybe I need to uninstall and reinstall?

**Comment by HighDeFing** [comment]
It works now, I managed to update docker destop for windows. I think it has to do with slow and unstable internet connections the download seems to freeze as well as the docker service, restarting and doing the reggit thing from here https://askubuntu.com/questions/1142359/error-0x80080005-server-execution-failed-while-installing-ubuntu-on-windows solved the issue. Maybe just restarting was enough, I just downloaded the update with multiple tries (the third one working). Also one shouldn't stop the service or restart it while downloading at least that's how I reproduce the error.
